### PR TITLE
update

### PR DESCRIPTION
**Pull request by arunkumar-patange**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -17,7 +17,8 @@ llama-index-readers-file==0.1.4
 langchain==0.1.12
 langchain-openai==0.0.8
 llama-index-llms-perplexity==0.1.3
+llama-index-llms-ollama
```
Comment (seyeong-han) [inline requirements.txt:20]: Thanks for adding the llama model!!

File: src/agent/base_agent.py [hunk 2]
```diff
@@ -116,7 +118,8 @@ def __init__(
 
         self.debug = debug
         self.routing_agent = ReActAgent.from_tools(
-            [search_tool, locate_tool, vision_tool], llm=llm, verbose=True
+            # [search_tool, locate_tool, vision_tool], llm=llm, verbose=True
+            [search_tool, locate_tool,], llm=llm, verbose=True
```
Comment (seyeong-han) [inline src/agent/base_agent.py:122]: Maybe we can make these tools with booleans to choose selectively.
Could you please add an argument for tool selection?

File: src/agent/base_agent.py [hunk 3]
```diff
@@ -168,15 +171,15 @@ def query(self, query: str) -> str:
         response = self.routing_agent.chat(query)
         self.routing_agent.reset()
         # write response to file for KG writeback
-        with open("data/external_response.txt", "w") as f:
+        with open("/brain/src/streamlit_app/data/external_response.txt", "w") as f:
```
Comment (seyeong-han) [inline src/agent/base_agent.py:174]: This absolute path is not acceptable for our repository.
How about setting `data_dir` argument for docker build?

File: src/agent/base_agent.py [hunk 4]
```diff
@@ -168,15 +171,15 @@ def query(self, query: str) -> str:
...
             print(response, file=f)
         # write back to the KG
         self.write_back()
         return response
 
     def write_back(self):
         documents = SimpleDirectoryReader(
-            input_files=["data/external_response.txt"]
+            input_files=["/brain/src/data/external_response.txt"]
```
Comment (seyeong-han) [inline src/agent/base_agent.py:182]: Same as the above comment.

File: src/agent/base_agent.py [hunk 5]
```diff
@@ -304,18 +307,18 @@ def get_routing_agent_response(self, query, return_entity=False):
         response = ""
         if self.debug:
             # writes ReAct agent steps to separate file and modifies format to be readable in .txt file
-            with open("data/routing_response.txt", "w") as f:
+            with open("/brain/src/streamlit_app/data/routing_response.txt", "w") as f:
```
Comment (seyeong-han) [inline src/agent/base_agent.py:310]: Same as the above comment.

File: src/agent/base_agent.py [hunk 6]
```diff
@@ -304,18 +307,18 @@ def get_routing_agent_response(self, query, return_entity=False):
...
                 orig_stdout = sys.stdout
                 sys.stdout = f
                 response = str(self.query(query))
                 sys.stdout.flush()
                 sys.stdout = orig_stdout
             text = ""
-            with open("data/routing_response.txt", "r") as f:
+            with open("/brain/src/streamlit_app/data/routing_response.txt", "r") as f:
```
Comment (seyeong-han) [inline src/agent/base_agent.py:317]: Same as the above comment.

File: src/agent/base_agent.py [hunk 7]
```diff
@@ -304,18 +307,18 @@ def get_routing_agent_response(self, query, return_entity=False):
...
                 text = f.read()
 
             plain = ansi_strip(text)
-            with open("data/routing_response.txt", "w") as f:
+            with open("/brain/src/streamlit_app/data/routing_response.txt", "w") as f:
```
Comment (seyeong-han) [inline src/agent/base_agent.py:321]: Same as the above comment.

File: src/agent/chat_agent.py [hunk 8]
```diff
@@ -59,3 +59,12 @@ def _replace_eks_to_from_message(self):
         self.message.llm_message[
             "knowledge_entity_store"] = self.entity_knowledge_store.get_memory(
             )
+
+
+class LLamaChatAgent(Agent):
+
+    def __init__(self, name, memory_stream_json, entity_knowledge_store_json,
+                 system_persona_txt, user_persona_txt, past_chat_json):
+        llm = Ollama('llama3-gradient')
```
Comment (seyeong-han) [inline src/agent/chat_agent.py:68]: Love it :+1:

File: streamlit_app/app.py [hunk 9]
```diff
@@ -34,6 +34,19 @@
...
+    memory_stream_json,
+    entity_knowledge_store_json,
+    system_persona_txt,
+    user_persona_txt,
+    past_chat_json,
+)
+
+
+chat_agent = openai_agent
+# chat_agent = llama_agent
```
Comment (seyeong-han) [inline streamlit_app/app.py:48]: Are you still working on `llama_agent`? You didn't use the `llama_agent`.

File: src/agent/base_agent.py [hunk 10]
```diff
@@ -58,6 +59,7 @@ def __init__(
         user_persona_txt,
         past_chat_json,
         debug=True,
+        llm = None,
```
Comment (seyeong-han) [inline src/agent/base_agent.py:62]: What about choosing llm model with a string argument?
So, we can get the proper agent using this `agent_instance = getattr(sys.modules[__name__], agent_name)()`.

File: streamlit_app/src [hunk 11]
```diff
@@ -0,0 +1 @@
+../src
```
Comment (seyeong-han) [inline streamlit_app/src:1]: I am sorry, but what is this for?

**Review by seyeong-han — CHANGES_REQUESTED** [review]
Thanks for all your effort!!
We really appreciate your contributing to the `docker build` and `make` scripts!
I left some of the comments that we hope you could change.
I think you are still working on this PR, right?

**Comment by seyeong-han** [comment]
If there is no activity on this PR, I will work on modifying all the requested features.
We really appreciate your docker work and the addition of Ollama. :100:

File: src/agent/base_agent.py (hunk 2 above)
Comment (shreybirmiwal) [reply src/agent/base_agent.py:122]: pr #26

**Comment by kingjulio8238** [comment]
Most implementations are included in PR #36 and in release v0.1.2.

**Pull request closed without merging**